x/sequencer/keeper/rotation.go

```diff
@@ -84,8 +85,17 @@ func (k Keeper) OnProposerLastBlock(ctx sdk.Context, proposer types.Sequencer) e
 
 	successor := k.GetSuccessor(ctx, rollapp)
 	k.SetProposer(ctx, rollapp, successor.Address)
+
 	k.SetSuccessor(ctx, rollapp, types.SentinelSeqAddr)
 
+	// if proposer is sentinel, prepare new revision for the rollapp
+	if successor.Sentinel() {
+		err := k.rollappKeeper.HardForkToLatest(ctx, rollapp)
```

Comment:
why do we need to hard fork here? can't we just hard fork when the new sequencer bonds? otherwise we gonna bump 2 revisions when the sequencer will come. 

if we don't hard fork here than already in the `RecoverFromHaltIfNeeded` should cover it no?